local/iatlas/cbioportal_export/clinical_to_cbioportal.py

```diff
@@ -0,0 +1,697 @@
+    "Riaz_Nivolumab_2017",
+    "VanAllen_antiCTLA4_2015",
+    "Zhao_NatMed_2019",
+]
+
+CBIOPORTAL_METADATA_COLS = [
+    "NORMALIZED_COLUMN_HEADER",
+    "ATTRIBUTE_TYPE",
+    "DATATYPE",
+    "DESCRIPTIONS",
+    "DISPLAY_NAME",
+    "PRIORITY",
+]
+
+CASE_LIST_TEXT_TEMPLATE = (
+    "cancer_study_identifier: {study_id}\n"
+    "stable_id: {stable_id}\n"
+    "case_list_name: {case_list_name}\n"
+    "case_list_description: {case_list_description}\n"
+    "case_list_ids: {case_list_ids}"
+)
+
+
+def preprocessing(
+    input_df_synid: str,
+    features_df_synid: str,
+    cli_to_cbio_mapping: pd.DataFrame,
+    cli_to_oncotree_mapping_synid: str,
+    datahub_tools_path: str,
+) -> pd.DataFrame:
+    """Preprocesses the data, runs the individual steps:
+        1. Gets the input clinical data
+        2. Gets the feature data and merged it in
+        3. Merges in the oncotree mappings
+        4. Remaps the columns to be cbioportal headers
+        5. Converts the oncotree codes to have the CANCER_TYPE and CANCER_TYPE_DETAILED columns
+        6. Splits the merged clinical data into patient and sample data
+
+    Args:
+        input_df_synid (str): Synapse id of input iatlas clinical dataset
+        features_df_synid (str): Synapse id of the features dataset
+        cli_to_cbio_mapping (pd.DataFrame): Clinical to cbioportal attirbutes mapping
+        cli_to_oncotree_mapping_synid (str): Oncotree mapping for clinical dataset
+        datahub_tools_path (str): Path to the datahub tools repo
+
+    Returns:
+        pd.DataFrame: preprocessed clinical merged dataset
+    """
+    input_df = pd.read_csv(syn.get(input_df_synid).path, sep="\t")
+    features_df = pd.read_csv(
+        syn.get(features_df_synid).path, sep="\t", low_memory=True
+    )
+    input_df = input_df.merge(features_df, how="left", on="sample_name")
+
+    cli_to_oncotree_mapping = pd.read_csv(
+        syn.get(cli_to_oncotree_mapping_synid).path, sep="\t"
+    )
+    oncotree_merge_cols = ["Cancer_Tissue", "TCGA_Study", "AMADEUS_Study"]
+
+    cli_with_oncotree = input_df.merge(
+        cli_to_oncotree_mapping[oncotree_merge_cols + ["ONCOTREE_CODE"]],
+        how="left",
+        on=oncotree_merge_cols,
+    )
+    cli_to_cbio_mapping_dict = dict(
+        zip(
+            cli_to_cbio_mapping["iATLAS_attribute"],
+            cli_to_cbio_mapping["NORMALIZED_HEADER"],
+        )
+    )
+    cli_remapped = cli_with_oncotree.rename(columns=cli_to_cbio_mapping_dict)
+    cli_remapped.rename(
+        columns={"sample_name": "SAMPLE_ID", "patient_name": "PATIENT_ID"}, inplace=True
+    )
+
+    cli_remapped.to_csv(
+        f"{datahub_tools_path}/add-clinical-header/cli_remapped.csv",
+        index=False,
+        sep="\t",
+    )
+    cli_w_cancer_types = convert_oncotree_codes(datahub_tools_path)
+    get_updated_cli_attributes(cli_to_cbio_mapping, datahub_tools_path)
+
+    return cli_w_cancer_types
+    
+    
+def split_into_patient_and_sample_data(
```

Comment:
It's a shame that GENIE as a package has a lot of red tape to contribute to, but these general tools could be an "easy to contribute" to package (imagine an overall "data tools" package.